storm-client/src/jvm/org/apache/storm/stats/StatsUtil.java

```diff
@@ -1589,18 +1570,16 @@ public static ComponentPageInfo aggCompExecsStats(
             if (lastReportedTime != null) {
                 reportedTime = lastReportedTime;
             } else {
-                reportedTime = 0;
+                reportedTime = lastReportedTime = 0;
```

Comment:
This change doesn't look like needed.

Reply:
we do not decide if updateHeartbeatCache is null when reportedTime.equals(lastReportedTime)

Reply:
I was confused that lastReportedTime is not used afterwards. My bad.